.github/workflows/pull_request_label_management.yml

```diff
@@ -0,0 +1,141 @@
+name: Pull Request Label Management
+
+on:
+  pull_request_target:
+    types: [labeled, unlabeled]
+
+env:
+  PAT_TOKEN: ${{ secrets.TEAM_READ_PAT }}  # PAT with read:org scopes
+
+permissions: {}
+
+jobs:
+  process-labels:
+    name: Process Label Actions
+    runs-on: ubuntu-22.04
+    permissions:
+      pull-requests: write
+
+    steps:
+      - name: Handle label changes
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ env.PAT_TOKEN }}
+          script: |
+            const { context, github } = require('@actions/github');
+            const octokit = github.rest;
+
+            // ======================
+            // GRAPHQL TEAM QUERIES
+            // ======================
+            async function getTeamMembers(teamSlug) {
+              const org = 'oppia';
+              try {
+                const result = await github.graphql({
+                  query: `
+                    query ($org: String!, $teamSlug: String!) {
+                      organization(login: $org) {
+                        team(slug: $teamSlug) {
+                          members(first: 100) {
+                            nodes { login }
+                          }
+                        }
+                      }
+                    }
+                  `,
+                  org,
+                  teamSlug,
+                  headers: {
+                    authorization: `token ${process.env.PAT_TOKEN}`
+                  }
+                });
+                
+                return new Set(
+                  result.organization.team.members.nodes
+                    .map(node => node.login.toLowerCase())
+                );
+              } catch (error) {
+                core.error(`Team query failed: ${error.message}`);
+                return new Set();
+              }
+            }
+
+
+            const action = context.payload.action;
+            const label = (context.payload.label?.name || '').trim();
+            const sender = (context.payload.sender?.login || '').toLowerCase();
+            const prNumber = context.issue.number;
+            const { owner, repo } = context.repo;
+
+            // Validate critical inputs
+            if (!/^[\w\s-:]+$/i.test(label)) {
+              core.error(`Invalid label format: ${label}`);
```

Comment:
This has the same problem as I flagged in #21828 here: https://github.com/oppia/oppia/pull/21828#discussion_r1957537786. Please check that all the lessons from #21828 have been applied here since the PRs are so similar